File: project-c-cosminpac04/motorcycleApp/Services/RaceService.cs
using System.Collections.Generic;
using motorcycleApp.Models;
using motorcycleApp.Repositories;

namespace motorcycleApp.Services
{
    public class RaceService
    {
        private readonly RaceRepository _raceRepository;

        public RaceService(RaceRepository raceRepository)
        {
            _raceRepository = raceRepository;
        }

        public void AddRace(Race race)
        {
            _raceRepository.Add(race);
        }

        public void RemoveRace(int id)
        {
            var race = _raceRepository.FindById(id);
            _raceRepository.Delete(race);
        }

        public void UpdateRace(int id, Race race)
        {
            _raceRepository.Update(race);
        }

        public Race GetRaceById(int id)
        {
            return _raceRepository.FindById(id);
        }

        public IEnumerable<Race> GetAllRaces()
        {
            return _raceRepository.GetAll();
        }
    }
}

File: project-java-cosminpac04/motorcycleContest/src/main/java/motorcycle/service/RaceService.java
package motorcycle.service;
import java.util.List;
import motorcycle.model.Race;
import motorcycle.repository.RaceHibernateRepository;
import motorcycle.repository.RaceRepository;


import java.util.List;

public class RaceService {
    private final RaceHibernateRepository raceRepository;

    public RaceService(RaceHibernateRepository raceRepository){
        this.raceRepository = raceRepository;
    }

    public void addRace(Race race){
        raceRepository.add(race);
    }

    public void deleteRace(Race race){
        raceRepository.delete(race);
    }

    public void updateRace(Race race, int id){
        raceRepository.update(race, id);
    }

    public Race findRaceById(int id){
        return raceRepository.findById(id);
    }

    public List<Race> getAllRaces(){
        return raceRepository.findAll();
    }
}
